client-v2/src/components/FrontsEdit/FrontsEditStrategyPicker.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { match } from 'react-router-dom';
+import Edit, { EditProps } from './Edit';
+import { Dispatch } from 'types/Store';
+import { connect } from 'react-redux';
+import getFrontsConfig from 'actions/Fronts';
+import {
+  fetchEditionsIssueAsConfig,
+  fetchFrontsConfig
+} from 'services/faciaApi';
+import { PriorityName } from 'types/Priority';
+
+type FrontsEditStrategyPicker = EditProps & {
+  match: match<{ priority: string; editionId?: string }>;
+  dispatch: Dispatch;
+};
+
+const isValidPathForEdition = (priority: string, id?: string): id is string =>
+  !!id && priority === 'edition';
+
+const isValidPathForOtherPriority = (priority: string, id?: string) =>
+  !id && priority !== 'edition';
+
+const getConfigThunk = (
+  dispatch: Dispatch,
+  priority: string,
+  editionId?: string
+) => {
+  if (isValidPathForEdition(priority, editionId)) {
+    return () =>
+      dispatch(getFrontsConfig(() => fetchEditionsIssueAsConfig(editionId)));
+  } else if (isValidPathForOtherPriority(priority, editionId)) {
+    return () => dispatch(getFrontsConfig(fetchFrontsConfig));
+  }
+  return null;
+};
+
+const FrontsEditStrategyPicker = ({
+  match: {
+    params: { priority, editionId }
+  },
+  dispatch,
+  ...props
+}: FrontsEditStrategyPicker) => {
+  const thunk = getConfigThunk(dispatch, priority, editionId);
```

Comment:
Just a thought -- we add our path to the application state, and we can create matchers from our route params; could this logic just exist in our thunk, avoiding the need for an extra component that doesn't have any presentational work to do?

Reply:
Oh yeah, I forgot we have that there ... yeah I think that could probably live outside there in the tree in that case 🆗